### PR TITLE
Enforce Single Attribute Per Line (#5501)

### LINKED ISSUE #5501: Change HTML/JSX formatting to have one attribute/prop per line
Similar to what was done for #3847 I think it best to break some of the discussion from #3101 into a new issue that people can 👍 or 👎.

I'm proposing that for HTML and JSX to have Prettier always have one attribute/prop per line (and thus not respect the developer's original preference). Code would then be formatted as shown below. This is in contrast to the current behaviour where we fit as much as possible with the print-width.

**Expected behavior:**
```jsx
<MyComponent lorem="1"/>

<MyComponent
  lorem="1"
  ipsum="2"
/>

<MyComponent
  lorem="1"
  ipsum="2"
  dolor="3"
/>
```

This suggestion for one attribute/prop per line was proposed several times in the discussion of #3101 but I think it is clearer if this is pulled into it's own proposal. The original proposal in #3101 is that Prettier would add an option to preserve original formatting which, while I agree with the author with the style that they want, I don't think a) an option, nor b) preserving original formatting follows the aims for Prettier (see also #2068). Instead I think the aims of #3101 are better served by this proposal to ignore the print-width and always place attributes/props on new lines (assuming that there is more than one).

This style appears to be the most common formatting for Angular, Vue and React from what I can tell. It style appears to be the style enforced by the rules:
* React: [jsx-max-props-per-line](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-max-props-per-line.md) which [Airbnb set to to 1 in their style guide](https://github.com/airbnb/javascript/blob/master/packages/eslint-config-airbnb/rules/react.js#L89)
* Vue: [max-attributes-per-line](https://github.com/vuejs/eslint-plugin-vue/blob/master/docs/rules/max-attributes-per-line.md) which has a default to 1.

### PR DESCRIPTION
**Pull request by kankje**
Added the `htmlSingleAttributePerLine` option for specifying if Prettier should enforce single attribute per line in HTML. Fixes #5501.

The implementation is based on this comment: https://github.com/prettier/prettier/issues/5501#issuecomment-441554159.

I checked out the [option philosophy](https://prettier.io/docs/en/option-philosophy.html) and it seems like this shouldn't be an option but I figured that the option would be better/safer for backward compatibility. I'd appreciate feedback on this though.

- [x] I’ve added tests to confirm my change works.
- [x] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [ ] (If the change is user-facing) I’ve added my changes to the `CHANGELOG.unreleased.md` file following the template.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by lydell** [comment]
If we add an option, shouldn’t it also affect Vue and JSX?

**Comment by evilebottnawi** [comment]
I think if we add option we should respect this option for all html like syntax

**Comment by kankje** [comment]
Hi, the option supports Vue and JSX the same way the `htmlWhitespaceSensitivity` option does. Should the option be renamed?

**Comment by lydell** [comment]
`htmlWhitespaceSensitivity` does not affect JSX. But you’re right that it does affect Vue.

Maybe `singleAttributePerLine`?

I only see tests for HTML?

**Comment by kankje** [comment]
I added support for JSX and added tests for both Vue and JSX.

**Comment by fisker** [comment]
I thought we don't accept those kind of options, remember how long takes to add `--vue-indent-script-and-style` ? https://github.com/prettier/prettier/issues/3888

**Comment by lydell** [comment]
We haven’t decided what to do yet. But if @kankje bothered to make a PR would could just as well make it nice. I think it helps discussion.

**Comment by kankje** [comment]
I'm fine with making this the default behavior if you feel like that's the way to do it.

**Comment by yippie** [comment]
Yes please, I think this is the way to go by default

**Comment by CaptainYouz** [comment]
hi everyone!
first of all thank you and nice job @kankje !
I noticed something on `.vue` files. If you have multiple attribute in your `<script>` or `<style>` tag inside your vue file, it also indent them one by line. For exemple
```
<style lang="scss" scoped>
``` 
become 
```
<style
  lang="scss"
  scoped
>
```
And strangely enough, in vsCode, it messes up with the editor colors (for the scss)(no big deal though).
It would be great if the rule is only applied to the template part of vue files. What do you think ?

**Comment by lydell** [comment]
@kankje This PR is good as-is. However, this will most likely stay open for quite a while as we decide what to do. First, we need to focus on getting 1.19 out.

#3101 and #5501 combined have the most :+1:s of all issues and is one of the least bike-sheddy option requests. So if we’re going to add another option, this one for sure has potential.

**Comment by JounQin** [comment]
Why not keep it like literal object?

If the developer write the first attribute in a new line, we should keep one attribute per line in the following, if he doesn't, we should keep it as is.

**Comment by SFWNSFW** [comment]
Hi, I'm using Prettier extensions in VS code, it seems this is the default settings now. Can some tell me how to config the json file to set this "htmlSingleAttributePerLine" option to false?

**Comment by lydell** [comment]
@SFWNSFW `htmlSingleAttributePerLine` does not exist – it’s just a proposal. If all attributes fits on one line, Prettier puts them on one line. If they don’t, Prettier puts one attribute on each line. One exception is if you have a `// comment` among the attributes forcing them to be multi-line. It’s always worked this way, so there must be something else going on in VS Code for you.

**Comment by dennishall** [comment]
> Why not keep it like literal object?
> 
> If the developer write the first attribute in a new line, we should keep one attribute per line in the following, if he doesn't, we should keep it as is.
> 
> Don't add such option please, we "hate" options.

--> because then it's not a unified rule, it's a segmented rule that behaves differently depending on the individual developer --- exactly what we're trying to avoid with tools like this.  However, I'm not opposed to making it a flavor of this rule: `maxAttributesPerLine: 1` OR `wrapAttributes: [many options, including "object-literal"] -- see js-beautify's `wrap-attributes` for example. (https://github.com/beautify-web/js-beautify#css--html)

**Comment by miszo** [comment]
Hi, any news regarding this PR? I really can't wait for it to be included in Prettier. @kankje, did you consider publishing this feature as a plugin until Prettier Team decide what to do with it?

**Comment by maogongzi** [comment]
@miszo Feel free to try out my fork at https://github.com/maogongzi/prettier/tarball/patch-single-line-attr-v0.1   ,simply replace the prettier version from your package.json to the following:

```
"prettier": "https://github.com/maogongzi/prettier/tarball/patch-single-line-attr-v0.1",
```

**Comment by diachedelic** [comment]
@maogongzi I have installed your fork but it does not seem to force attributes onto their own line (`fill` and `d` should be on separate lines)

```
$ npx prettier src/icons/Add.vue 
<template>
  <svg
    class="xc-icon xc-add"
    viewBox="2 2 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path fill="none" d="M0 0h24v24H0V0z" />
  </svg>
</template>

$ npx prettier --version
1.20.0-dev

// yarn.lock
prettier@^1.18.2, "prettier@https://github.com/maogongzi/prettier/tarball/patch-single-line-attr-v0.1":
  version "1.20.0-dev"
  resolved "https://github.com/maogongzi/prettier/tarball/patch-single-line-attr-v0.1#56db4cff79930f505538c4ba25507a92727b82a6"
```

**Comment by maogongzi** [comment]
@diachedelic Yes I understand, that's perhaps your .eslintrc needs some tiny tweaks, it really takes skill to get Vue, ESlint and Prettier tuned, I'll send you a copy of my `.prettierrc`, `.eslintrc`, and `.editorconfig` so that you can compare with your own config to figure out what's happening, for me, the `svg` you pasted out is formatted like below, each attr stays on it's own line:

```
      <!-- dialog title -->
      <div class="dialog__header clearfix">
        <svg
          class="xc-icon xc-add"
          viewBox="2 2 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            fill="none"
            d="M0 0h24v24H0V0z"
          ></path>
        </svg>
        <h3>
          <template v-if="mode === 'create'">
```

My config files pasted here:

`.editorconfig`
https://pastebin.com/JBPD4UeC

`.prettierrc`
https://pastebin.com/BnkzDAsm

`.eslintrc`
https://pastebin.com/8iSHG4DW

Have a good day!

**Comment by diachedelic** [comment]
ah yes I did not see the `singleAttributePerLine` for `.prettierrc`, thanks

**Comment by NoPhaseNoKill** [comment]
Are we able to merge this? I'm hanging out for this to be added

**Comment by j-f1** [comment]
@NoPhaseNoKill You could try using `prettier-eslint`.

**Comment by NoPhaseNoKill** [comment]
@j-f1 Having a look at their docs, it's unclear how I'd make this work inside of my .eslintrc.js and .prettierrc.js files. The only example they give is this. Any ideas on what that'd look like?

```
const format = require("prettier-eslint");

// notice, no semicolon in the original text
const sourceCode = "const {foo} = bar";

const options = {
  text: sourceCode,
  eslintConfig: {
    parserOptions: {
      ecmaVersion: 7
    },
    rules: {
      semi: ["error", "never"]
    }
  },
  prettierOptions: {
    bracketSpacing: true
  },
  fallbackPrettierOptions: {
    singleQuote: false
  }
};

const formatted = format(options);

// notice no semicolon in the formatted text
formatted; // const { foo } = bar
```

**Comment by NoPhaseNoKill** [comment]
Looks like I want this instead: https://github.com/prettier/prettier-eslint-cli . This will offer me a way to do it via the CLI (or in my case, through husky pre-commit script). This allows me define the config paths for prettier and eslint. Thanks for the help @j-f1. This seems like a pretty reasonable solution to this issue for those waiting/those who don't want to fork in the meantime

**Comment by izznogooood** [comment]
Can someone please get "kraken" on resolving these conflicts? :)

Great work guys. This will be much appreciated!

**Comment by thedavefulton** [comment]
I'm curious if this kind of functionality could be added as a plugin. I looked over the plugin docs and the tutorial and it seems like plugins are for adding (or replacing?) languages rather than adding an option to an existing language. Am I reading that right or can someone point me to an example of an option plugin?

If it is possible I'd be happy to take a swing at it as this is really the only feature keeping us from adopting prettier. Thanks, be well 🖖

**Comment by kankje** [comment]
I fixed the conflicts and updated the "first available in" version to 2.1.0.

**Comment by Vadorequest** [comment]
> @kankje This PR is good as-is. However, this will most likely stay open for quite a while as we decide what to do. First, we need to focus on getting 1.19 out.
> 
> #3101 and #5501 combined have the most 👍s of all issues and is one of the least bike-sheddy option requests. So if we’re going to add another option, this one for sure has potential.

It's saddening to see that such MUST HAVE features are slow down in such a way by  people who obviously don't use the related tools (JSX, here).

It's great to have an opinion and to limit the number of option to keep things simple, but wasting months of discussion about something that's really needed for code quality, code review and **PRODUCTIVITY** is just disappointing to see.

I'm barely getting started with Prettier, but this is a blocking requirement.
I will not enforce a company-wide tool like prettier if it enforces the wrong rules, period. (especially without any workaround)

https://github.com/prettier/prettier/issues/3101 has been open in 2017, v2 is out, maybe the time to merge this PR has come?

**Comment by jasek** [comment]
Please merge 🙏

**Comment by Vadorequest** [comment]
Sorry for the spam, but I've open https://github.com/prettier/prettier/issues/8507 and I believe all people who are interested in this PR could have something constructive to say about how Prettier could do better about code quality.

Its goal is also to help the community understand what are the motivations of the Prettier core maintainers.

**Comment by thorn0** [comment]
> are slow down in such a way by people who obviously don't use the related tools (JSX, here).

@Vadorequest Prettier is heavily used in Facebook, the company that created JSX. If you could stop writing baseless speculations like that, that'd be great. Thanks.

**Comment by jzfed** [comment]
I felt like the project was being controlled by a bunch of old fogies who didn't understand how to think about the needs of most users and only think about their own habits. The project is sorry for its name, the code isn't prettier but uglier, and it also wastes a lot of time trying to figure out how to solve this simple problem with an option. If so, we would be better off using other plugins rather than wasting time with these old fogies.

Goodbye, uglier.

**Comment by andrewjrhill** [comment]
Any updates on when this will be merged? This is a pretty vital change for any modern front-end framework (or library) which is heavily reliant on HTML'ish attributes.

**Comment by Vadorequest** [comment]
@andrewjrhill I fear/believe it won't happen. I suggest you take a look at https://github.com/prettier/prettier/issues/8507 to understand better Prettier own's motivations regarding this kind of changes.

**Comment by andrewjrhill** [comment]
@Vadorequest - That is rather unfortunate given only 8% of the total "thumbs" on this pull request are against the change. If @thorn0 is looking for a strong argument as to why this change should (or should not) be merged; all he has to do is look at the numbers.

It seems strange to draw the line on the proposed `htmlSingleAttributePerLine` option, yet allow for similar options such as `jsxBracketSameLine`. I fail to understand the logic behind ignoring a contribution that would benefit a large portion of the development community.

**Comment by lydell** [comment]
> It seems strange to draw the line on the proposed htmlSingleAttributePerLine option, yet allow for similar options such as jsxBracketSameLine

We mention this exact thing in the [docs](https://prettier.io/docs/en/option-philosophy.html):

> But others are harder to motivate in hindsight, and usually end up with bike shedding. --arrow-parens, --jsx-single-quote, --jsx-bracket-same-line and --no-bracket-spacing are not the type of options we want more of. They exist (and are difficult to remove now), but should not motivate adding more options like them.

**Comment by andrewjrhill** [comment]
@lydell - Fair enough.

Much to my dismay; I would recommend closing this pull request as https://prettier.io/docs/en/option-philosophy.html makes it clear that Prettier is opinionated and is not open to additional options.

**Comment by egemon** [comment]
@kankje 
Thanks for the work!
I just used ur branch locally and used .prettierrc.js as config and 
`singleAttributePerLine: true,`
bit it didn't work really for JSX. 
Am I missing smth, is this branch still works 4 u?

**Comment by dkovacevic15** [comment]
Splitting component props over multiple lines is such a win for readability that I can't believe anyone would, in good conscience, call this request bike shedding.

**Comment by spaceneenja** [comment]
Bump. Please consider merging this.

The reasons for including this feature are extensive, the drawbacks minimal. We already have freaking option for indentation on style and script tags and this is far more impactful to readability.

If we have to kill one to gain one, I choose this one 100/100 times.

**Comment by Vadorequest** [comment]
@dkovacevic15 @spaceneenja I tried to discuss this in https://github.com/prettier/prettier/issues/8507 
 (summary [here](https://github.com/prettier/prettier/issues/8507#issuecomment-650519899)) and it got closed in favor of https://github.com/prettier/prettier/issues/8507 where nothing new happened yet.

In short, I feel like maintainers don't know how to implement such changes without breaking the Prettier vision, and it could be implemented through a plugin system. But it's really theoretical so far.

I think adding some proposal/comment there (https://github.com/prettier/prettier/issues/8507) may speed up things, or at least move forward on this issue.

**Comment by lydell** [comment]
> I tried to discuss this in #8507 (summary here) and it got closed in favor of #8507

@Vadorequest You wrote 8507 both times :)

**Comment by izznogooood** [comment]
Since we are so many people who wants this. Obviously of different skill and profession.

Would anyone be willing to discuss a fork, and I don't mean a new direction, but basing on prettier adding options? Closely following prettier.

**Comment by Vadorequest** [comment]
> Since we are so many people who wants this. Obviously of different skill and profession.
> 
> Would anyone be willing to discuss a fork, and I don't mean a new direction, but basing on prettier adding options? Closely following prettier.

I'm interested. I actually tried to fork it myself, but got lost in the complexity to be honest. 

My initial thought was to fork, then merge a few PR regarding stuff that matters to me and let it be. I don't feel like Prettier needs a lot of updates or fixes regarding the software itself, since it tries to do as little changes as possible.

Having someone familiar with how Prettier works on the inside (e.g: a maintainer) would help tremendously, because there are a lot of repos at https://github.com/prettier and it was unclear to me of how they relate together.

**Comment by lydell** [comment]
I think you only need to fork the main `prettier/prettier` repo. Then, you can install using `"prettier": "npm:prettier-awesome-fork@^1.2.3"` to make all tooling pick it up.

**Review by kasunsewwandika — APPROVED** [review]

**Comment by danielehrhardt** [comment]
Could someone merge this one?

**Comment by glen-84** [comment]
Since JSX isn't HTML, `xmlSingleAttributePerLine` might be a better name for the reasons that I mentioned [here](https://github.com/prettier/prettier/issues/5377#issuecomment-689393993):

> - JSX is apparently "JavaScript XML".
> - HTML in Prettier is formatted as XHTML (as far as I know?).

**Comment by egorovsa** [comment]
Any news? Is anybody going to finish this PR?

**Comment by andrewjrhill** [comment]
This rule not being merged is the only thing stopping us from adopting prettier. It's a shame really.

**Comment by jd-0001** [comment]
Sorry to say but I think I also have to stop using prettier due to this. I hope they merge it ASAP.

**Comment by FrankySnow** [comment]
I also stopped using Prettier in favour of [`js-beautify-html`](https://github.com/beautify-web/js-beautify#css--html) for the same reasons, and I'm pretty happy with it. I might consider turning back to Prettier once this issue is resolved.

**Comment by Romanior** [comment]
I think we can relax here, main reason why this never gonna be merged is mentioned above:
https://github.com/prettier/prettier/pull/6644#issuecomment-641302412
Facebook invested in this tool and they would never change if something does not fit their workflow.

**Comment by alexander-akait** [comment]
@Romanior Nope, most of contributors don't work in facebook

**Comment by narayanpai** [comment]
Looks like lot of folks are hovering around for this PR to be merged. Prettier would be showered with ❤️  is someone merges this PR

**Comment by izznogooood** [comment]
What's Prettier ?

**Review by bhlibovytskyi — APPROVED** [review]

**Comment by bhlibovytskyi** [comment]
@kankje Hi there, is it going to merged some day?

**Comment by andrewjrhill** [comment]
@bhlibovytskyi - Over 100 thumbs up and the prettier maintainers have essentially decided to just "ignore" this pull request instead of dealing with it once and for all. It's been in limbo for almost 2 years and they refuse to merge and refuse to close it either. The mind boggles.

**Comment by narayanpai** [comment]
@bhlibovytskyi Can you help merging this PR?

**Comment by scherii** [comment]
image

@narayanpai I think you are referring to @bhlibovytskyi's approval from 8 days ago?
These reviews can be done by anyone in the [«Files changed» tab](https://github.com/prettier/prettier/pull/6644/files).

image

So my (educated 😄) guess is that only some of [the people listed here](https://github.com/orgs/prettier/people) can help move this issue forward.

**Comment by kapsiR** [comment]
@alexander-akait @azz @BPScott @CiGit @dangmai @duailibe @fisker @ikatyang @j-f1 @jlongster @kaicataldo @kddeisz @kentcdodds @lipis

Any statement would be helpful 😉

**Comment by kddeisz** [comment]
@kapsiR what are you doing? I don't even work on core I work on plugins. Going out of your way to purposefully add to already full inboxes when you don't even know if folks are involved is rude at best.

**Comment by kapsiR** [comment]
I'm very sorry to have bothered you with this, but I can understand the displeasure about this issue and just wanted to help with it a bit.

**Comment by egorovsa** [comment]
O mine god.. Still nothing ? It's seems out of my mind to get why this too tricky to merge it.

**Comment by CalebKester** [comment]
> @kapsiR what are you doing? I don't even work on core I work on plugins. Going out of your way to purposefully add to already full inboxes when you don't even know if folks are involved is rude at best.

Hi @kankje , I'm very new to prettier's plugin development. I thought about reaching out to you directly but figured this ticket is the best place for it as it's related. Regardless of what happens to this ticket, do you think this functionality is something that a plugin can do? It looks like prettier's plugins can only add new languages and not enhance existing? Is there a way to import a language and modify it?

**Comment by jodaka** [comment]
For one a half year we've been writing `"prettier": "https://github.com/maogongzi/prettier/tarball/patch-single-line-attr-v0.1",` in our package.json files. I feel guilty about it. 

Is there any way I can help with this PR? It seems like original PR author has lost any interest :-(

**Comment by wrslatz** [comment]
Could I pick this PR up and help get it across the line (at minimum resolve conflicts)? cc @kankje

**Comment by alexander-akait** [comment]
It need to be rebased also we need `singleAttributePerLine` and respect it for all html syntaxes

**Comment by kankje** [comment]
Hi and sorry for not being active on the discussion, I've been busy with a lot of other stuff after the last update. I can have a look at rebasing this on the weekend. I also noticed that the following note has been added to the option philosophy: "Option requests aren’t accepted anymore". Should I just remove the option and make this the default behavior?

**Comment by alexander-akait** [comment]
Let's keep option right now

**Comment by maogongzi** [comment]
> For one a half year we've been writing `"prettier": "https://github.com/maogongzi/prettier/tarball/patch-single-line-attr-v0.1",` in our package.json files. I feel guilty about it.
> 
> Is there any way I can help with this PR? It seems like original PR author has lost any interest :-(

LOL, glad to hear that my fork helped you, also I have worked my ass off to apply @kankje 's patch on the `v2.3.0` mainline version, I have a PR, UT is failing but I tested on my project and looks like it works, PR is: 

https://github.com/maogongzi/prettier/pull/7/files 

If you wish to create your own fork and fix UTs to ensure everything works fine, you could mainly focus on the changes I made to those files(and thanks to @kankje again :)

```
src/language-html/options.js
src/language-html/printer-html.js
src/language-js/options.js
src/language-js/print/jsx.js
```

The question is, based on my test, `v2.3.0` based fork doesn't offer me enough motivation to upgrade, yes there might be some optimizations on formatting, but to me for now it's merely a "nice-to-have", instead of "must-have" since I'm not a perfectionist on this topic, on the same time, some "BREAKING CHANGES" have been introduced, based on my tests, there are three "BREAKING CHANGES" in formatting in v2.3.0:

1. object properties will stay in the same line if the line length hasn't reach the limitation

   for example you'll have something like this: ` let { x, y, z } = blabla;` instead of

   ```
   let {
     x,
     y,
     z
   } = blabla;
   ```

2. the same as object properties, css classes will also be merged into the same line if there is still room on the line.

   so you'll have something like `<div class="foo bar blabla"></div>` instead of

   ```
   <div
     class="
       foo
       bar
       blabla
     "
   ></div>
   ```

3. also, short chained method callings will be "compressed" into one line

   so you'll have something like `a().b().c()` instead of

   ```
   a()
     .b()
     .c();
   ```

Anyway, I have scheduled to upgrade to this new fork in sometime in the future, but not now :D

Wish you a good day!

**Comment by kankje** [comment]
Alright, I rebased the branch to `main` and pushed the changes again.

**Comment by duola8789** [comment]
@maogongzi 
我使用了 singleAttributePerLine 之后， vue组件中的style标签也会被一并处理了，您有遇到这个情况吗
image

**Comment by maogongzi** [comment]
> @maogongzi
> 我使用了 singleAttributePerLine 之后， vue组件中的style标签也会被一并处理了，您有遇到这个情况吗
> image

这个是预期结果，我这边也是这样 :)

**Comment by egorovsa** [comment]
Seems prettier's team is not going to add this feature.it ridiculous however it's a fact!  let's go away...

**Comment by CalebKester** [comment]
Could we just maintain the fork of it here with instructions on how to use it in our own projects? I appreciate all the work that's been done to get it this far and just wonder if we can use it on the side and keep it updated.

**Comment by yuwenhuang** [comment]
It's going to be 2022 in 1.5 months and this is still not merge yet...

**Comment by pjg** [comment]
@kankje any chance for you to rebase this branch against prettier's main branch?

**Comment by kankje** [comment]
> @kankje any chance for you to rebase this branch against prettier's main branch?

I've done that a few times now, is there any guarantee that this is going to be merged if I do that? 😅

**Comment by pjg** [comment]
@kankje Pretty sure the answer is "no" to your question. It's more about us, your minions, who depend on and swear by your fork but would like to pull latest from the canonical prettier as well (i.e. the best of both worlds).

**Comment by kankje** [comment]
> @kankje Pretty sure the answer is "no" to your question. It's more about us, your minions, who depend on and swear by your fork but would like to pull latest from the canonical prettier as well (i.e. the best of both worlds).

Alright, I'll have a look at doing that on the weekend, _my minions_.

**Comment by jd-0001** [comment]
@lydell Can't we merge it as this is requested by so many people?

Also, this is [recommended](https://eslint.vuejs.org/rules/max-attributes-per-line.html) by Vue linting rules.

I personally do not use prettier because I prefer single attributes per line, If this PR gets merged I will love using prettier just like before.

People from Vue community are eagerly waiting for this (according to my thoughts) and commented a lot in this thread.

I really like ❤️  to use prettier in every project so with due respect can you please merge it? @kankje is already rebasing it and maintained the PR up to date till now.

It will be great if the Prettier community can show some love here :)

Regards.

**Comment by alexander-akait** [comment]
@prettier/core What do you think? Maybe we should revisit it, I am against new options, but there are a lot of requests and I don't see it unreadable, also there are a lot of requests

**Comment by egorovsa** [comment]
@kankje u r the best Man!

**Comment by pjg** [comment]
@kankje thank you very much, sir! 🎉  🎉  🎉

**Comment by egorovsa** [comment]
> @prettier/core What do you think? Maybe we should revisit it, I am against new options, but there are a lot of requests and I don't see it unreadable, also there are a lot of requests

Dear @prettier/core The same question. It's the third time when @kankje is resolving conflicts. It would be quite respectful in regards to him to notice his work!

**Comment by fisker** [comment]
I'm fine adding this option, though I won't use it myself.

**Review by duailibe — APPROVED** [review]

**Comment by jodaka** [comment]
So who's gonna be that Hero who pushes Merge button?

**Comment by alexander-akait** [comment]
I hope I didn't offend anyone by doing this :fearful:

**Comment by egorovsa** [comment]
My goodness. It happened! Thanks !

**Comment by alexander-akait** [comment]
We took too long to decide on this, it's time to move forward, this is a fairly popular request that I think we should satisfy

**Comment by liquidvisual** [comment]
I don't see the new option in the [docs](https://prettier.io/docs/en/options.html)?

**Comment by miszo** [comment]
> I don't see the new option in the [docs](https://prettier.io/docs/en/options.html)?

I guess that it's not released yet.

**Comment by miszo** [comment]
@sosukesuzuki, 

This is disturbing: https://github.com/prettier/prettier/commit/44bffcdc5cac94904340ee2887171679d0a13423

So, you're going to revert it? Or it is just for some release purposes and it will be re-reverted?

**Comment by sosukesuzuki** [comment]
@miszo To release new patch version. Please read the description of https://github.com/prettier/prettier/pull/11906. I'm sorry for making you uncomfortable.

**Comment by miszo** [comment]
@sosukesuzuki,

No worries 🙂 
Maybe that was wrong wording from my side - I was more confused than disturbed 😅 

I understand - thanks for the clarification anyways and good luck with releasing pathc version! 🤞

**Comment by fisker** [comment]
I didn't see a changelog, and it says `_First available in v2.3.3_`.

**Comment by fisker** [comment]
@kankje  ^

**Comment by kankje** [comment]
> I didn't see a changelog, and it says `_First available in v2.3.3_`.

@fisker What would you like me to do? Should I open a new PR that updates the "first available in" version to 2.6.0 and adds a changelog to `changelog_unreleased/html`?

**Comment by fisker** [comment]
Yes, please

**Comment by kankje** [comment]
Opened a PR: https://github.com/prettier/prettier/pull/11934

**Comment by MilesHeise** [comment]
I see this getting added and then reverted (and a big convo about version numbers in the linked revert PR) . . . is there a best place to check/subscribe to find out when this feature is actually available? for accessibility reasons my team is totally blocked on adding Prettier to HTML so I have been following this issue eagerly for a couple years now and definitely psyched to see it might finally be coming :-D 

definitely want to upgrade versions the second this makes it in . . .

**Comment by scherii** [comment]
Hi @MilesHeise 

I suggest you keep an eye out for the blog post for version 2.6:
https://prettier.io/blog/

Now that the work is done, it's just wait and see.
And rest assured that I am at least as eager for this release. 😉

**Comment by MilesHeise** [comment]
excellent, thanks for those details!

**Comment by sosukesuzuki** [comment]
This has been released! https://prettier.io/blog/2022/03/16/2.6.0.html

**Comment by chriszrc** [comment]
Was there ever a discussion of being able to automatically alphabetize the attributes as well? I really want this :)

**Comment by JounQin** [comment]
> Was there ever a discussion of being able to automatically alphabetize the attributes as well? I really want this :)

It is out of scope of `prettier` core itself.

https://prettier.io/docs/en/rationale.html#what-prettier-is-_not_-concerned-about

You can try to implement a prettier plugin by yourself.

**Pull request merged**
